scripts/oligomap_output_to_sam_nh_filtered.py

```diff
@@ -0,0 +1,450 @@
+#!/usr/bin/env python
+
+"""Transform oligomap output FASTA file to SAM keeping the best alignments.
+
+Read the input FASTA file, pick the best alignment(s) for each read (i.e. all
+of the alignments have either 0 or 1 error). In addition, if the `--nh-filter`
+CLI argument is set, filter the reads with more hits than the number provided.
+The following sections will cover what the FASTA file must be like and how the
+output can look like.
+
+EXPECTED INPUT
+The FASTA file must be the output of mapping your library with the tool
+oligomap: "https://github.com/zavolanlab/oligomap". The alignment
+representation has to have 6 lines with the following content:
+- The first line must contain the read's name, its length, the start and end
+mapped positions in the read's sequence, the reference sequence name and, the
+start and end mapped positions in the reference sequence in this order.
+- The second line, must contain the reference sequence name.
+- The third line, must contain the fields "errors:" and "orientation:" with
+the alignment number of errors and reference sequence strand.
+- The fourth and sixth lines must be the read and reference sequence
+respectively.
+- The fifth line must contain a vertical bar for each match between the
+sequences base pairs and a space otherwise.
+
+Alignments must be separated with a blank line.
+In addition, alignments have to be sorted by the read name and separated
+with a blank line. An example of how an entry would look like in the EXAMPLES
+section.
```

Comment:
It would actually make more sense to provide this information on the `oligomap` project page rather than here.

Reply:
I'll put this information in the main README of the `oligomap` repository and redirect the user of this script there.

Reply:
Perfect. Please resolve the comment after you are done.